chainer/functions/pooling/average_pooling_nd.py

```diff
@@ -0,0 +1,148 @@
+        gx /= functools.reduce(operator.mul, self.ksize)
+        return gx,
+
+    def backward_gpu(self, x, gy):
+        if (cuda.cudnn_enabled and self.use_cudnn and
+                pooling_nd._check_cudnn_acceptable_type(x[0].dtype)):
+            # With cuDNN v3 or greater, use cuDNN implementation for inputs
+            # with spatial dimensions of two or more.
+            if _cudnn_version >= 3000 and self.ndim >= 2:
+                return super(AveragePoolingND, self).backward_gpu(x, gy)
+            # With cuDNN v2, use cuDNN implementation only for inputs with
+            # spatial dimensions of two.
+            elif self.ndim == 2:
+                return super(AveragePoolingND, self).backward_gpu(x, gy)
+
+        n, c = x[0].shape[:2]
+        dims = x[0].shape[2:]
+        ys = gy[0].shape[2:]
+        gx = cuda.cupy.empty_like(x[0])
+        coeff = 1. / functools.reduce(operator.mul, self.ksize)
+
+        in_params, out_params, operation, name = \
+            average_pooling_nd_kernel.AveragePoolingNDKernelBackward.generate(
+                self.ndim)
+        cuda.elementwise(in_params, out_params, operation, name)(
+            gy[0].reduced_view(),
+            *(dims + ys + self.ksize + self.stride + self.pad + (coeff, gx)))
+
+        return gx,
+
+    def create_pool_desc(self):
+        return cudnn.create_pooling_descriptor(
+            self.ksize, self.stride, self.pad,
+            libcudnn.CUDNN_POOLING_AVERAGE_COUNT_INCLUDE_PADDING)
+
+
+def average_pooling_nd(x, ksize, stride=None, pad=0, use_cudnn=True):
+    """N-dimensionally spatial average pooling function.
+
+    This function provides a N-dimensionally generalized version of
+    :func:`~functions.average_pooling_2d`. This acts similarly to
+    :class:`~functions.ConvolutionND`, but it computes the average of input
+    spatial patch for each channel without any parameter instead of computing
+    the inner products.
+
+    Args:
+        x(~chainer.Variable): Input variable.
+        ksize (int or tuple of ints): Size of pooling window. ``ksize=k`` and
+            ``ksize=(k, k, ..., k)`` are equivalent.
+        stride (int or tuple of ints or None): Stride of pooling applications.
+            ``stride=s`` and ``stride=(s,s, ..., s)`` are equivalent. If
+            ``None`` is specified, then it uses same stride as the pooling
+            window size.
+        pad (int or tuple of ints): Spatial padding width for the input array.
+            ``pad=p`` and ``pad=(p, p, ..., p)`` are equivalent.
+        use_cudnn (bool): If ``True`` and cuDNN is enabled, then this function
+            uses cuDNN as the core implementation. cuDNN supports more than
+            one-dimensional pooling.
+
+    Returns:
+        ~chainer.Variable: Output variable.
+
+    .. note::
+
+       This function currently does not support ``cover_all`` mode as
+       :func:`max_pooling_nd`. Average pooling runs in non-cover-all mode.
```

Comment:
Is there some technical reason why cover-all mode is not supported? Or is it simply because you could not spare enough time?